src/plugins/customer-order/generate-order/generate-order.resolver.ts

```diff
@@ -0,0 +1,112 @@
+        let skus: Array<String> = []
+        jsonOrder.map((sku: any) => {
+            skus.push(sku.code);
+        });
+        if (skus.length == 0) {
+            throw new UserInputError('Solicitud incorrecta!');
+        }
+        let search: Array<ProductVariant>;
+        try {
+            search = await this.generateCustomerOrderService.findProductVariantsBySKU(ctx, skus);
+        } catch (e) {
+            search = [];
+        }
+
+        let order: Order | any | undefined;
+
+        if (search.length > 0 && this.generateCustomerOrderService.isStockAvailable(search)) {
+            try{
+                order = await this.orderService.getActiveOrderForUser(ctx, ctx.session.user?.id as ID);
+                if (!order) {
+                    order = await this.orderService.create(ctx,ctx.session.user?.id as ID);
+                }
+            }catch(e){console.log("Error to get or create order");}
+            if (!order) {
+                throw new InternalServerError("Ocurrió un error al intentar genear el pedido");
+            }
+
+            try{
+                for await (let[index, productVariant] of search.entries()) {
+                    let jsonValue = jsonOrder.find((order: any) => order.code == productVariant.sku)
+                    // se verifica que se pueda asignar el producto, de lo contrario omitimos agregar el producto a la orden
+                    if (jsonValue && (productVariant.stockOnHand - productVariant.stockAllocated) > 0) {
+                        try{
+                            let addItemResult: ErrorResultUnion<Order, any> = await this.orderService.addItemToOrder(ctx, order?.id as ID, productVariant.id, jsonValue.quantity);
+                            if (addItemResult instanceof Order){
+                                order = addItemResult;
+                            } else if(addItemResult.order){;
+                                order = addItemResult.order
+                            }
+                        }catch(e){/** error to add products to order beacuse insuficient stock  */}
+                    }
+                };
+            }catch(e) {console.log(`ERROR ADDING ITEMS TO ORDER, detail: ${e}`);}
+
+            if (order) {
+                if (order.lines.length > 0) {
+                    try {
+                        let transitionResult = await this.orderService.transitionToState(ctx, order.id, 'ArrangingPayment');
+                    } catch (e) {
+                        try{
+                            await this.orderService.cancelOrder(ctx,{orderId: order.id,reason: "ERROR on transi¡tion state to ArrangingPayment" });
+                        } catch(e){}
+                        throw new IllegalOperationError("Ocurrio un error al procesar el pedido");
+                    }
+
+                    try {
+                        let paymentMethods = await this.orderService.getEligiblePaymentMethods(ctx, order.id);
```

Comment:
No es necesario usar `let`, puede ser `const`
```suggestion
                        const paymentMethods = await this.orderService.getEligiblePaymentMethods(ctx, order.id);
```